src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -206,6 +204,10 @@ Copyright (c) .NET Foundation. All rights reserved.
    <Target Name="PrepareForILLink"
            DependsOnTargets="_ComputeManagedAssemblyToLink">
 
+    <!-- We print a message to the user to explain that trimming is 
+         potentially problematic when warnings are suppressed. -->
+    <NETSdkInformation Condition="'$(PublishTrimmed)' == 'true' And '$(SuppressTrimAnalysisWarnings)' == 'true'" ResourceName="ILLinkOptimizedAssemblies" />
```

Comment:
I think you can remove the PublishTrimmed condition since this target only kicks in when trimming.